btk/metrics.py

```diff
@@ -747,31 +747,33 @@ def __next__(self):
                         f"number of surveys: {len(surveys)}, instead len is {len(meas_band_num)}"
                     )
                 metrics_results_f = {}
-                for i, surv in enumerate(blend_results["isolated_images"].keys()):
+                for i, surv_name in enumerate(blend_results["isolated_images"].keys()):
                     additional_params = {
-                        "psf": blend_results["psf"][surv],
-                        "pixel_scale": surveys[i].pixel_scale,
+                        "psf": blend_results["psf"][surv_name],
+                        "pixel_scale": surveys[i].pixel_scale.to("arcsec").value,
                         "meas_band_num": meas_band_num[i],
                         "verbose": self.verbose,
                     }
+                    band_name = surveys[i].available_filters[meas_band_num[i]]
```

Comment:
Seems a bit overkill to retrieve the filter name. I guess there should be a better way.

Reply:
the easiest would be to replace `meas_band_num` with the band name directly I think. What do you think @thuiop is that feasible in `metrics.py` ?

Reply:
We do not really use the band name except to recover the corresponding filter. However, we do need the meas_band_num at some point, to access the images in that band.

Reply:
Oh yeah good point, let's keep it as is then. Unless @aboucaud has an alternative suggestion?

Reply:
would have to go deeper in the code for proper suggestion. another time